app/scripts/metamask-controller.js

```diff
@@ -61,6 +61,21 @@ module.exports = class MetamaskController extends EventEmitter {
     // network store
     this.networkController = new NetworkController(initState.NetworkController)
 
+    // setup onLine lister for restarting the provider
+    if (this.platform.addOnLineListener) {
```

Comment:
"online" is one word, I'd rather we didn't camel case it into two. Non blocking.